cron_status.py

```diff
@@ -0,0 +1,244 @@
+        'container_name',
+        'current_status',
+        'overall_status',
+        'current_msg',
+        'changed',
+    ])
+    current_status_list = []
+    changes_to_report = False
+    for container_name, report in status_history_list[0].items():
+        # get known history of this container
+        container_status_history = []
+        container_status_history_filtered = []
+        for entry in status_history_list:
+            if container_name in entry:
+                status = entry[container_name][0]
+                # strip out STARTING (starting is as good as its result)
+                container_status_history.append(status)
+                if status == ContainerStatus.STARTING:
+                    continue
+                container_status_history_filtered.append(status)
+
+        overall_status = container_status_history[0]
+        # TODO: if the status changes too often, stop reporting changes
+        # and switch to FLAPPING state
+        # if flapping:
+        #     overall_status = ContainerStatus.FLAPPING
+
+        if len(container_status_history_filtered) > 2:
+            # enough history data is available.
+            # we want report a change for this container if the current status
+            # different to the oldest status we know
+            changed = container_status_history_filtered[0] != container_status_history_filtered[1]
+        else:
+            # not enough history is available. always report failure.
+            changed = container_status_history[0] == ContainerStatus.FAILED
+
+        # if there is at least one change, we want to report changes
+        changes_to_report = changes_to_report or changed
+
+        current_status_list.append(ExtendedStatusReport(
+            container_name, report[0], overall_status, report[1], changed
+        ))
+
+    return changes_to_report, current_status_list
+
+
+def main():
+    """Update HTML page and send emails on status changes."""
+    os.makedirs(STATUS_DIR, exist_ok=True)
+    # get old and new status json
+    status_report_list_new, stderr, returncode = get_new_status()
+    status_history_list = get_old_status()
+    status_history_list.insert(0, status_report_list_new)
+    # detect flapping
+    changes_to_report, current_status_list = detect_flapping_and_changes(status_history_list)
+    current_status_list = sorted(current_status_list)
+    # convert json to text/html and update html page
+    title = '%s kastenwesen status' % socket.getfqdn()
+    content_html = format_status(current_status_list, out_format='html')
+    content_text = format_status(current_status_list, out_format='ascii')
+    update_html_page(content_html, stderr, title)
+    # send mail if needed
+    if returncode == 42:
+        # another instance is running.
+        # Never send mails during that.
+        # Also do not save the status history to make sure that no relevant change mails are suppressed.
+        exit()
```

Comment:
As I asked in #56:

> Question: other_instance_running always seems to be False because, shell is in read_only_args. Am I right? So we can remove this logic. I also think that test won't/shouldn't fail even if another instance of kastenwesen is running...

Do we still need this?